external/config-checker/src/main/java/org/wso2/security/tools/utils/Utils.java

```diff
@@ -0,0 +1,601 @@
+                throw new ConfigCheckerException("Unable to create an instance of the class " + entry.getValue(), e);
+            }
+        }
+
+        if (configCheckerReportGenerator == null) {
+            throw new ConfigCheckerException("unable to find an report generator for the given report format "
+                    + configCheckerReportOutFormat);
+        }
+        return configCheckerReportGenerator;
+    }
+
+    /**
+     * Get the appropriate parser according to the given file format.
+     *
+     * @param fileFormat
+     * @return
+     * @throws ConfigCheckerException
+     */
+    public static ConfigurationParser getConfigurationParser(String fileFormat) throws ConfigCheckerException {
+        ConfigurationParser configurationParser = null;
+        Map<String, String> configurationParserMap = ConfigCheckerSettings.getInstance()
+                .getParsers();
+
+        for (Map.Entry<String, String> entry : configurationParserMap.entrySet()) {
+            try {
+                if (entry.getKey().equals(fileFormat)) {
+                    configurationParser = (ConfigurationParser) Class.forName(entry.getValue()).newInstance();
+                }
+            } catch (ClassNotFoundException e) {
+                throw new ConfigCheckerException("Unable to find the class " + entry.getValue(), e);
+            } catch (InstantiationException | IllegalAccessException e) {
+                throw new ConfigCheckerException("Unable to create an instance of the class " + entry.getValue(), e);
+            }
+        }
+
+        if (configurationParser == null) {
+            throw new ConfigCheckerException("unable to find an parser for the given file format "
+                    + fileFormat);
+        }
+        return configurationParser;
+    }
+
+    /**
+     * This method is used to get the list of files in a given product package.
+     *
+     * @param filename
+     * @param productDirectoryPath
+     * @return a list of file paths inside the given directory path with the given file name.
+     */
+    public static List<String> searchConfigFiles(String filename, String productDirectoryPath)
+            throws ConfigCheckerException {
+        List<String> paths = new ArrayList<>();
+        try {
+            Files.walk(Paths.get(productDirectoryPath))
+                    .filter(Files::isRegularFile)
+                    .forEach((f) -> {
+                        String filePath = f.toString();
+                        File file = new File(filePath);
+                        if (file.getName().equals(filename)) {
+                            paths.add(filePath);
+                        }
+                    });
+        } catch (IOException e) {
+            throw new ConfigCheckerException("IO Exception occurred while retrieving the configuration files", e);
+        }
+        return paths;
+    }
+
+    /**
+     * This method is used to get the configuration file name.
```

Comment:
This description and the method name are contradicting. Shall we improve the comment?